src/cloud/audiocom/ShareAudioDialog.cpp

```diff
@@ -0,0 +1,842 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  ShareAudioDialog.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#include "ShareAudioDialog.h"
+
+#include <wx/bmpbuttn.h>
+#include <wx/button.h>
+#include <wx/clipbrd.h>
+#include <wx/gauge.h>
+#include <wx/stattext.h>
+#include <wx/statline.h>
+#include <wx/textctrl.h>
+
+#include "AllThemeResources.h"
+#include "BasicUI.h"
+#include "MemoryX.h"
+#include "Project.h"
+#include "ShuttleGui.h"
+#include "Theme.h"
+#include "Track.h"
+#include "TempDirectory.h"
+
+#include "ServiceConfig.h"
+#include "OAuthService.h"
+#include "UploadService.h"
+#include "UserService.h"
+
+#include "LinkAccountDialog.h"
+#include "UserImage.h"
+
+#include "CodeConversions.h"
+
+#include "export/Export.h"
+#include "ui/AccessibleLinksFormatter.h"
+
+#include "widgets/HelpSystem.h"
+
+#ifdef HAS_CUSTOM_URL_HANDLING
+#include "URLSchemesRegistry.h"
+#endif
+
+#ifdef USE_LIBFLAC
+extern ChoiceSetting FLACBitDepth;
+extern ChoiceSetting FLACLevel;
+#endif
+
+#ifdef USE_WAVPACK
+extern IntSetting QualitySetting;
+extern IntSetting BitrateSetting;
+extern IntSetting BitDepthSetting;
+extern BoolSetting HybridModeSetting;
+#endif
```

Comment:
Let it stand for alpha, but I don't like these `extern`s and I hope you don't either.  It makes the dialog dependent on src/export files but the dependency is not made explicit in an `include` directive.  Those export files are header-less because they are supposed to be highest-level things in the dependency graph, and candidates for easy removal as modules, if we can ever return to the extraction refactorings.

I'm sure you or I will think up something in the future to rewrite the necessary parts of this file as a registry that the exporters can plug into.  Making possible tinier builds of Audacity that only know Wav exporting for instance.

Reply:
I don't like it, but I don't think that it is feasible to rewrite the whole Export framework. It is highly coupled with the UI and provides pretty much no sane way to configure the exporting plugin.